airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/AsyncStreamConsumer.java

```diff
@@ -156,6 +157,14 @@ public void accept(final String messageString, final Integer sizeInBytes) throws
       validateRecord(message);
 
       getRecordCounter(message.getRecord().getStreamDescriptor()).incrementAndGet();
+    } else if (Type.STATE.equals(message.getType())) {
+      if (message.getState().getType() != null && message.getState().getType() == STREAM) {
+        final StreamDescriptor streamDescriptor = message.getState().getStream().getStreamDescriptor();
+        if (Strings.isNullOrEmpty(streamDescriptor.getNamespace())) {
```

Comment:
do we have a test that verifies we still checkpoint correctly on streams with null namespace after this change? AsyncStreamConsumerTest seems (at a glance) to only have tests with nonnull namespace

(my guess is that checkpointing is broken after this, but I have no evidence :shrug:)

Reply:
Why do we need to override the namespace here?

Reply:
So I made the change to remove this mutation. Take a look at my comment in class `GlobalAsyncStateManager`
```
/**
   * If the user has selected the Destination Namespace as the Destination default while setting up
   * the connector, the platform sets the namespace as null in the StreamDescriptor in the
   * AirbyteMessages (both record and state messages). The destination checks that if the namespace is
   * empty or null, if yes then re-populates it with the defaultNamespace. See
   * {@link io.airbyte.cdk.integrations.destination_async.AsyncStreamConsumer#accept(String,Integer)}
   * But destination only does this for the record messages. So when state messages arrive without a
   * namespace and since the destination doesn't repopulate it with the default namespace, there is a
   * mismatch between the StreamDescriptor from record messages and state messages. That breaks the
   * logic of the state management class as {@link descToStateIdQ} needs to have consistent
   * StreamDescriptor. This is why while trying to extract the StreamDescriptor from state messages,
   * we check if the namespace is null, if yes then replace it with defaultNamespace to keep it
   * consistent with the record messages.
   */
```